src/crl/interactivesessions/_process.py

```diff
@@ -249,8 +250,17 @@ class _BackgroundProcessWithoutPty(_BackgroundProcessBase,
 class _NoCommBackgroudProcess(_BackgroundProcessWithoutPty):
 
     def run(self):
-        self._initialize_terminal()
-        return self._get_cmd_pid()
+        with self._in_terminal():
+            self._initialize_terminal()
```

Comment:
This line (254) could be moved to the beginning of context manager.

Reply:
Done